### PR TITLE
chore(error/dx): proper naming

### PR DESCRIPTION
**Pull request by hyf0**

### CODE REVIEW

**Comment by hyf0** [comment]
* **#2501** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2501?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2500** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2500?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2499** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2499?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2498** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2498?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hyf0 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/rolldown_error/src/type_aliases.rs [hunk 1]
```diff
@@ -1,6 +1,10 @@
 use crate::BuildDiagnostic;
 
-pub type DiagnosableResult<T> = Result<T, Vec<BuildDiagnostic>>;
+pub type BatchedBuildDiagnostic = Vec<BuildDiagnostic>;
+
+pub type UnaryBuildResult<T> = std::result::Result<T, BuildDiagnostic>;
```
Comment (IWANABETHATGUY) [inline crates/rolldown_error/src/type_aliases.rs:5]: Can you change another naming? `unary` more likely used in the mathematical domain, 
**一元构建结果** sounds wired,  I used both google translate and deepL

**Review by IWANABETHATGUY — APPROVED** [review]
looks good to me except the naming.

**Comment by hyf0** [comment]
### Merge activity

* **Oct 24, 12:42 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/rolldown/rolldown/2500) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

File: crates/rolldown_error/src/type_aliases.rs (hunk 1 above)
Comment (hyf0) [reply crates/rolldown_error/src/type_aliases.rs:5]: `UnaryBuildResult ` is only used in a few places, so I give `BuildResult` the name to `Vec<BuildDiagnostic>`.

I used to name them `BuildResult` and `BatchedBuildResult`, but changed after. It's welcome to suggesting other naming.

Comment (IWANABETHATGUY) [reply crates/rolldown_error/src/type_aliases.rs:5]: Maybe `SingleBuildResult`? just pure English translation, https://www.perplexity.ai/search/qing-gei-wo-yi-ge-ming-ming-ji-Gsy4Upk_T3GHc55rXf3tFw#1

Comment (hyf0) [reply crates/rolldown_error/src/type_aliases.rs:5]: Let's consider the current naming is also temporary and keep open for other naming.

Comment (hyf0) [reply crates/rolldown_error/src/type_aliases.rs:5]: > Maybe `SingleBuildResult`? just pure English translation, https://www.perplexity.ai/search/qing-gei-wo-yi-ge-ming-ming-ji-Gsy4Upk_T3GHc55rXf3tFw#1

Yeah. Pretty straightforward. Let me change to this.

**Pull request merged**